### PR TITLE
[feat] - Add support for react-native apple sign in.

### PR DESCRIPTION
**Pull request by matheusccastroo**
Create new handler for login with react-native. While this was made thinking of react-native, it allows any front-end to use the package and correctly handle login, given that they handle the client-side credential fetching from apple.

### CODE REVIEW

File: apple_server.js [hunk 1]
```diff
@@ -67,8 +67,9 @@ const verifyAndParseIdentityToken = (query, idToken, isNative = false) =>
  * @param query
  * @param {*} tokens tokens and data from apple
  * @param isNative
+ * @param fromLoginHandler
  */
-const getServiceDataFromTokens = (query, tokens, isNative = false) => {
+const getServiceDataFromTokens = (query, tokens, isNative = false, fromLoginHandler = false) => {
```
Comment (renanccastro) [inline apple_server.js:72]: this name `fromLoginHandler` should be more descriptive.

```suggestion
const getServiceDataFromTokens = (query, tokens, isNative = false, isBeingCalledFromLoginHandler = false) => {
```

**Review by renanccastro — CHANGES_REQUESTED** [review]
small change, but overall lgtm

**Review by renanccastro — CHANGES_REQUESTED** [review]
lgtm, but please update the readme with the new handler so we can have an up to date documentation. also an use case example would be good.

File: apple_server.js [hunk 2]
```diff
@@ -67,8 +67,9 @@ const verifyAndParseIdentityToken = (query, idToken, isNative = false) =>
  * @param query
  * @param {*} tokens tokens and data from apple
  * @param isNative
+ * @param isBeingCalledFromLoginHandler
  */
-const getServiceDataFromTokens = (query, tokens, isNative = false) => {
+const getServiceDataFromTokens = (query, tokens, isNative = false, isBeingCalledFromLoginHandler = false) => {
```
Comment (filipenevola) [inline apple_server.js:72]: it would be better to change this to an options object. So we don't need to rely on order.

File: apple_server.js [hunk 3]
```diff
@@ -269,8 +272,11 @@ const getServiceData = query =>
...
 Accounts.registerLoginHandler(query => {
-  if (query.methodName != 'native-apple') {
+  const methodName = query.methodName;
+  if (!["native-apple", "web-apple"].includes(methodName)) {
     return;
   }
-  return getServiceDataFromTokens(query, getTokens(query, true), true);
-});
+
+  const isNative = methodName === "native-apple";
```
Comment (filipenevola) [inline apple_server.js:280]: this string should be a constant.

**Review by filipenevola — CHANGES_REQUESTED** [review]

**Pull request merged**